src/conn/conn_api.c

```diff
@@ -190,6 +190,42 @@ __wt_conn_remove_collator(WT_SESSION_IMPL *session)
 }
 
 /*
+ * __conn_compat_config --
+ *	Configure compatibility version.
+ */
+static int
+__conn_compat_config(WT_SESSION_IMPL *session, const char **cfg)
+{
+	WT_CONFIG_ITEM cval;
+	WT_CONNECTION_IMPL *conn;
+
+	conn = S2C(session);
+	WT_RET(__wt_config_gets(session, cfg,
+	    "compatibility.major", &cval));
+	if (cval.val) {
+		if (cval.val <= WIREDTIGER_VERSION_MAJOR)
+			conn->compat_major = (uint16_t)cval.val;
+		else
+			WT_RET_MSG(session, EINVAL, "unknown major version");
+	} else
+		conn->compat_major = WIREDTIGER_VERSION_MAJOR;
+	WT_RET(__wt_config_gets(session, cfg,
+	    "compatibility.minor", &cval));
+	/*
+	 * We don't currently have knowledge of major/minor pairs.  If the
+	 * user gave us zero set it to the current version.  Check for some
+	 * out of bounds settings.
+	 */
+	conn->compat_minor = (uint16_t)cval.val;
```

Comment:
Per our phone conversation, I think it's worth exploring to see how the string version ends up, but I'm not wedded to the suggestion.